src/main/java/seedu/address/model/person/DonatedAmount.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the amount donated by a donor in thousands of dollars (k dollar).
+ * Guarantees: immutable; is valid as declared in {@link #isValidAmount(String)}
+ */
+public class DonatedAmount {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Donated amount should be a non-negative number with up to two decimal places.";
+
+    /*
+     * The validation regex allows only non-negative numbers with up to two decimal places.
+     */
+    public static final String VALIDATION_REGEX = "\\d+(\\.\\d{1,2})?";
+
+    private final Double amount;
+
+    /**
+     * Constructs a {@code DonatedAmount}.
+     *
+     * @param amount A valid non-negative number representing the donated amount.
+     */
+    public DonatedAmount(Double amount) {
+        requireNonNull(amount);
+        checkArgument(isValidAmount(amount.toString()), MESSAGE_CONSTRAINTS);
```

Comment:
Just a small suggestion. Would passing in `amount` instead of a `String` be more accurate in this case